twill-core/src/main/java/org/apache/twill/internal/DefaultResourceReport.java

```diff
@@ -158,4 +174,31 @@ public String toString() {
       ", services=" + services +
       '}';
   }
+
+  private TwillRunResources wrapTwillRunResources(final String runnableName, TwillRunResources resources) {
```

Comment:
The `logLevels` for all running containers are stored in `ResourceReport`, and `ResourceReport` have a map containing the `TwillRunResources` for each running runnable. We want to return the correct log levels for each runnable. So we wrap the `TwillRunResources` with `ForwardingTwillRunResources`, which has a delegation and we override the `getLogLevels` methods to get the log levels from the map in `ResourceReport`.